### ISSUE TITLE
Bot stops working sometimes

### ISSUE THREAD

**Issue by Aleksaas** [CuteOne/BadRotations#16]
Sometimes in fight bot stops casting spells for 5-10 seconds, and then continue normally. Did anyone else also notice this?

**Comment by Vilt-Coding** [comment]
Class/spec?

**Comment by Aleksaas** [comment]
Fury warrior.

**Comment by Vilt-Coding** [comment]
Ashley does warrior stuff, might be an engine issue.
Do you get any (lua) errors? Or anything else that might identify the issue?

**Comment by Aleksaas** [comment]
I didnt notice any lua errors. If I see some pattern when this problem occurs I will tell you.

**Comment by CuteOne** [comment]
I have been leveling my warrior, hadnt personally noticed anything :( Any more direct details you can give would be helpful

**Comment by Aleksaas** [comment]
I have been researching through code to try to figure out what was causing this problem, and I noticed something in Core.lua. The `function BadRotationsUpdate(self)` is called every 1/10 of second, but what if the next call starts before previous one finished? It can happen. So I added something like
`    if br.updateInProgress ~= true then
        self.updateInProgress = true
        // Code goes here
        self.updateInProgress = false
    end
`
Also I changed that `br.pulse:makeEnTable()` wasnt called on every pusle but on COMBAT_LOG_EVENT_UNFILTERED. I also commented out call of `br.pulse:ttd()` because I dont need it. And some other changes for fury warrior. After these changes I noticed that my FPS went up, previously I had problems with that. And also for 2 days now I didnt have a problem with bot stop working... I am not sure if those changes helped for that problem to be fixed but it is just a suggestion for you what maybe can cause problem...
